### ISSUE TITLE
Submission: tidypmc

### ISSUE THREAD

**Issue by cstubben** [ropensci/software-review#290]
Submitting Author: Chris Stubben (@cstubben)  
Repository:  https://github.com/cstubben/tidypmc
Version submitted:   1.1
Editor: @noamross 
Reviewer 1: @toph-allen  
Reviewer 2: @juliasilge  
Archive: TBD  
Version accepted: TBD   

---

-   Paste the full DESCRIPTION file inside a code block below:

```
Package: tidypmc
Type: Package
Title: Parse full text XML documents from PMC
Version: 1.1
Author: Chris Stubben
Maintainer: Chris Stubben <chris.stubben@hci.utah.edu>
License: GPL-3
URL: https://github.com/cstubben/tidypmc
BugReports: https://github.com/cstubben/tidypmc/issues
Encoding: UTF-8
VignetteBuilder: knitr
Depends: xml2
Imports:
    europepmc,
    tokenizers,
    stringr,
    tibble,
    dplyr,
    readr
Suggests:
    tidytext,
    knitr
Description: This package parses section paragraphs, captions, tables, references and metadata from XML documents in the Open Access subset of Pubmed Central. Additional functions are available to search text and expand ranges of the references cited, locus tags and gene operons.
RoxygenNote: 6.1.1
```

## Scope 

- Please indicate which category or categories from our [package fit policies](https://ropensci.github.io/dev_guide/policies.html#package-categories) this package falls under: (Please check an appropriate box below. If you are unsure, we suggest you make a pre-submission inquiry.):

	- [ ] data retrieval
	- [ ] data extraction
	- [ ] database access
	- [x] data munging
	- [ ] data deposition
	- [ ] reproducibility
	- [ ] geospatial data
	- [x] text analysis
	

- Explain how and why the package falls under these categories (briefly, 1-2 sentences):
`tidypmc` parses paragraphs, captions, tables and references in XML documents from Pubmed Central into tidy formats for searching and further analysis with `tidytext`

-   Who is the target audience and what are scientific applications of this package?  
Researchers that want to analyze full text documents in PMC. 
-   Are there other R packages that accomplish the same thing? If so, how does yours differ or meet [our criteria for best-in-category](https://ropensci.github.io/dev_guide/policies.html#overlap)?
`pubchunks`.   `pub_chunks(x, "body")` returns a list with an s3 class and this output can be converted to a data.frame, but the document structure is lost and text often contains nested tables, mathML and other text blobs.  `pmc_text` removes tables and other tags [incorrectly] nested in //sec/p tags and tries to preserve the document structure by saving the full path to subsection titles, paragraph numbers and sentence in a tibble.   

-   If you made a pre-submission enquiry, please paste the link to the corresponding issue, forum post, or other discussion, or @tag the editor you contacted.

## Technical checks

Confirm each of the following by checking the box.  This package:

- [x] does not violate the Terms of Service of any service it interacts with. 
- [x] has a CRAN and OSI accepted license.
- [x] contains a [README with instructions for installing the development version](https://ropensci.github.io/dev_guide/building.html#readme). 
- [x] includes [documentation with examples for all functions](https://ropensci.github.io/dev_guide/building.html#documentation).
- [x] contains a vignette with examples of its essential functions and uses.
- [ ] has a [test suite](https://ropensci.github.io/dev_guide/building.html#testing).
- [ ] has [continuous integration](https://ropensci.github.io/dev_guide/ci.html), including reporting of test coverage using services such as Travis CI, Coveralls and/or CodeCov.

## Publication options

- [x] Do you intend for this package to go on CRAN?  
- [ ] Do you wish to automatically submit to the [Journal of Open Source Software](http://joss.theoj.org/)? If so:

<details>
 <summary>JOSS Options</summary>  

  - [ ] The package has an **obvious research application** according to [JOSS's definition](https://joss.readthedocs.io/en/latest/submitting.html#submission-requirements).
    - [ ] The package contains a `paper.md` matching [JOSS's requirements](https://joss.readthedocs.io/en/latest/submitting.html#what-should-my-paper-contain) with a high-level description in the package root or in `inst/`.
    - [ ] The package is deposited in a long-term repository with the DOI: 
    - (*Do not submit your package separately to JOSS*)  
  
</details>

- [ ] Do you wish to submit an Applications Article about your package to [Methods in Ecology and Evolution](http://besjournals.onlinelibrary.wiley.com/hub/journal/10.1111/(ISSN)2041-210X/)? If so:  

<details>
<summary>MEE Options</summary>  

- [ ] The package is novel and will be of interest to the broad readership of the journal. 
- [ ] The manuscript describing the package is no longer than 3000 words.
- [ ] You intend to archive the code for the package in a long-term repository which meets the requirements of the journal (see [MEE's Policy on Publishing Code](http://besjournals.onlinelibrary.wiley.com/hub/journal/10.1111/(ISSN)2041-210X/journal-resources/policy-on-publishing-code.html))
- (*Scope: Do consider MEE's [Aims and Scope](http://besjournals.onlinelibrary.wiley.com/hub/journal/10.1111/(ISSN)2041-210X/aims-and-scope/read-full-aims-and-scope.html) for your manuscript. We make no guarantee that your manuscript will be within MEE scope.*)
- (*Although not required, we strongly recommend having a full manuscript prepared when you submit here.*)
- (*Please do not submit your package separately to Methods in Ecology and Evolution*)

</details>

## Code of conduct

- [x] I agree to abide by [rOpenSci's Code of Conduct](https://ropensci.github.io/dev_guide/policies.html#code-of-conduct) during the review process and in maintaining my package should it be accepted.

**Comment by melvidoni** [comment]
Hello @cstubben,  thank you for your submission.

On the Scope section of the opening post, could you explain possible differences and similitudes between your package and  [pubchunks](https://github.com/ropensci/pubchunks/) which is already onboarded to rOpenSci?

**Comment by cstubben** [comment]
Thanks for the suggestion.  pubchunks has now been added.

**Comment by melvidoni** [comment]
Thank you for submitting @cstubben! @noamross has been assigned as your editor.

**Comment by noamross** [comment]
### Editor checks:

- [x] **Fit**: The package meets criteria for [fit](policies.md#fit) and [overlap](policies.md#fit)
- [ ] **Automated tests:** Package has a testing suite and is tested via Travis-CI or another CI service.
- [x] **License:** The package has a CRAN or OSI accepted license
- [x] **Repository:** The repository link resolves correctly
- [ ] **Archive** (JOSS only, may be post-review): The repository DOI resolves correctly
- [ ] **Version** (JOSS only, may be post-review): Does the release version given match the GitHub release (v1.0.0)?

---

#### Editor comments

Hello @cstubben, thanks for your submission! This package is a good fit under our Aims and Scope.  However, we require that packages have test suite with good coverage of package functionality and continuous integration before we pass them on to review.  You can find more details here: <https://ropensci.github.io/dev_guide/building.html#testing>

In addition, here's the output of `goodpractice::gp()` run on your package.  You should address these notes as well before we move to review.

Please let me know if you have questions. Once you have addressed these issues we can move to assigning reviewers. 

```
── GP tidypmc ──────────────────────────────────

It is good practice to

  ✖ write unit tests for all functions, and all package code
    in general. 0% of code lines are covered by test cases.

    R/collapse_rows.R:20:NA
    R/collapse_rows.R:21:NA
    R/collapse_rows.R:22:NA
    R/collapse_rows.R:23:NA
    R/collapse_rows.R:24:NA
    ... and 398 more lines

  ✖ not use "Depends" in DESCRIPTION, as it can cause name
    clashes, and poor interaction with other packages. Use "Imports"
    instead.
  ✖ use '<-' for assignment instead of '='. '<-' is the
    standard, and R users and developers are used it and it is easier
    to read your code for them if you use '<-'.

    R/separate_refs.R:20:11

  ✖ avoid long code lines, it is bad for readability. Also,
    many people prefer editor windows that are about 80 characters
    wide. Try make your lines shorter than 80 characters

    R/pmc_caption.R:28:1
    R/pmc_caption.R:29:1
    R/pmc_caption.R:31:1
    R/pmc_caption.R:32:1
    R/pmc_caption.R:35:1
    ... and 81 more lines

  ✖ avoid sapply(), it is not type safe. It might return a
    vector, or a list, depending on the input data. Consider using
    vapply() instead.

    R/collapse_rows.R:29:13
    R/pmc_caption.R:28:13
    R/pmc_caption.R:32:13
    R/pmc_caption.R:35:16
    R/pmc_caption.R:36:16
    ... and 31 more lines

  ✖ avoid 1:length(...), 1:nrow(...), 1:ncol(...),
    1:NROW(...) and 1:NCOL(...) expressions. They are error prone and
    result 1:0 if the expression on the right hand side is zero. Use
    seq_len() or seq_along() instead.

    R/collapse_rows.R:40:32
    R/pmc_caption.R:48:62
    R/pmc_caption.R:67:59
    R/pmc_caption.R:111:39
    R/pmc_caption.R:116:62    ... and 12 more lines

  ✖ not import packages as a whole, as this can cause name
    clashes between the imported packages. Instead, import only the
    specific functions you need.
```

**Comment by noamross** [comment]
Hi @cstubben, I just wanted to check in to see if you had any questions about the above and if you are planning to continue with this submission.

**Comment by cstubben** [comment]
Yes, I'm still trying to figure out the best way to do the unit tests.  I'd like to loop through 10K articles in one of the XML files at [Europe PMC FTP](https://europepmc.org/ftp/oa) rather than just test a single article if that's possible.

**Comment by noamross** [comment]
Good to hear.  If you have questions about topics like testing, the rOpenSci [forum](https://discuss.ropensci.org/) is a good place to ask.

**Comment by cstubben** [comment]
I just updated the package and fixed the `gp()` suggestions except two.  I added tests for all functions but still get warnings?   Also, I would like to leave the dependency to `xml2` since the package parses an `xml_document` and users should know how to view nodes, for example queries like this are highly recommended: `cat(as.character(xml_find_all(doc, "//ref")[1]))`.  More details on xml2 parsing are in the vignette.

Here's the lastest `gp()`

```
It is good practice to

 ✖ write unit tests for all functions, and all package code in general. 77% of code lines are covered by
   test cases.

   R/collapse_rows.R:22:NA
   R/collapse_rows.R:36:NA
   R/collapse_rows.R:40:NA
   R/collapse_rows.R:46:NA
   R/path_string.R:23:NA
   ... and 116 more lines

 ✖ not use "Depends" in DESCRIPTION, as it can cause name clashes, and poor interaction with other
   packages. Use "Imports" instead.
 ✖ not import packages as a whole, as this can cause name clashes between the imported packages.
   Instead, import only the specific functions you need.
```

**Comment by noamross** [comment]
Thank you, quite an improvement!  A few notes:

- The "loading packages as a whole message" is a matter of judgement - we just try to avoid importing a whole package when only a few functions are needed.  I will leave it to reviewers to say if there are specific dependencies that can be streamlined.
- re: `xml2`, even if you use it in a vignette and its complementary to your package, you should still leave the package in `Imports` (or `Suggest` if you don't depend on it otherwise) and leave it to the user to load it to use those functions.
- 77% test coverage is reasonable.  We ask the reviewers to inspect the package tests to ensure that appropriate functionality is tested.  However, this is also one reason why it is we require that you have continuous integration with test status and coverage badges set up - so reviewers can inspect the results.  See this chapter in our guide for instructions: https://ropensci.github.io/dev_guide/ci.html .  Once this is done we can proceed.

**Comment by cstubben** [comment]
Ok, I fixed the Imports issue and added Travis CI and coverage badge.

**Comment by noamross** [comment]
Thank you! I am now seeking reviewers.

**Comment by noamross** [comment]
Reviewers are @toph-allen and @juliasilge, thank your for agreeing to review! Due date for reviews is 2019-05-02.

@cstubben, you can also our peer-review badge to your repo README:
```
[![](https://badges.ropensci.org/290_status.svg)](https://github.com/ropensci/software-review/issues/290)
```

**Comment by cstubben** [comment]
Thanks, I added that badge and a second vignette.  I don't plan to make any more changes until reviews are returned

**Comment by noamross** [comment]
👋@toph-allen and @juliasilge, just a friendly reminder that you're reviews are due this coming 
Thursday, 2019-05-02.

**Comment by toph-allen** [comment]
I've got a start on mine here: [toph-allen/tidypmc-review ](https://github.com/toph-allen/tidypmc-review) (trying out the `pkgreviewr` package for this one). I should have no problem getting it done by Thursday.

**Comment by juliasilge** [comment]
I know this is a little late, but I am going to get this done this coming weekend. You'll have it by Monday!

**Comment by toph-allen** [comment]
Here's my review! Additionally, the output of `pkgreviewr`'s index.Rmd notebook, including the results of tests and stuff, is viewable [here](https://toph-allen.github.io/tidypmc-review/index.nb.html). Nice work, @cstubben!

@noamross & @cstubben, please let me know if there are any further questions about what I've written.

---

## Package Review

*Please check off boxes as applicable, and elaborate in comments below.  Your review is not limited to these topics, as described in the reviewer guide*

- [x] As the reviewer I confirm that there are no [conflicts of interest](#coi) for me to review this work (If you are unsure whether you are in conflict, please speak to your editor _before_ starting your review).

#### Documentation

The package includes all the following forms of documentation:

- [ ] **A statement of need** clearly stating problems the software is designed to solve and its target audience in README
- [x] **Installation instructions:** for the development version of package and any non-standard dependencies in README
- [x] **Vignette(s)** demonstrating major functionality that runs successfully locally
- [x] **Function Documentation:** for all exported functions in R help
- [x] **Examples** for all exported functions in R Help that run successfully locally
- [ ] **Community guidelines** including contribution guidelines in the README or CONTRIBUTING, and DESCRIPTION with `URL`, `BugReports` and `Maintainer` (which may be autogenerated via `Authors@R`).

In summary: Looks good for the most part! Installation instructions, vignettes, and documentation for exported functions are present. I think some minor edits to the README, giving just a little more by way of introduction and context to the package's use cases, and adding contribution guidelines, would help it conform a bit more to the packaging guidelines.

The DESCRIPTION's `Title` — and in general, wherever that sort of introductory piece of text is used without context, e.g. the `R/tidypmc-package.R` help` — should spell out "PubMed Central" rather than using the acronym "PMC".

The GitHub Issues page is given for `BugReports` in the DESCRIPTION, but there are no contributing guidelines in the README or a separate CONTRIBUTING file.

The documentation for all the core functions is comprehensive and contains examples, both in the package's help and in the README file.

The README gives a comprehensive walkthrough of the package's features and examples of their usage, and points the user to the vignettes for further information. I think it could do with a slightly longer introduction, though. Per the first item in the above list, it could describe more of the high-level goals that a user could accomplish with it, or explain a little more how it might fit into users' workflows.

For example, it could describe why a user might want to parse an XML file from the PMC OAS, and why this package is better at that than a generic XML parser.

While the vignettes explain how to work on raw XML files, the README only mentions using the `europepmc` package, and I initially wondered if this was the only way it could read files (I experimented with `xml2` before I discovered the documentation in the vignettes). It might be good to just note, where the README mentions using `europepmc`, that users can also read XML files directly using `xml2`, and refer the user to the vignettes there.

The "Introduction to tidypmc" vignette describes how the package finds and replaces superscript references — which is really clever — but skimming this documentation, it sounded like the package didn't do this automatically. Perhaps move the sentence saying that `pmc_text()` adds brackets to superscript references to the start of that section of the vignette.

>##### For packages co-submitting to JOSS
>
>- [ ] The package has an **obvious research application** according to [JOSS's definition](http://joss.theoj.org/about#submission_requirements)
>
>The package contains a `paper.md` matching [JOSS's requirements](http://joss.theoj.org/about#paper_structure) with:
>
>- [ ] **A short summary** describing the high-level functionality of the software
>- [ ] **Authors:**  A list of authors with their affiliations
>- [ ] **A statement of need** clearly stating problems the software is designed to solve and its target audience.
>- [ ] **References:** with DOIs for all those that have one (e.g. papers, datasets, software).

#### Functionality

- [x] **Installation:** Installation succeeds as documented.
- [x] **Functionality:** Any functional claims of the software been confirmed.
- [x] **Performance:** Any performance claims of the software been confirmed.
- [x] **Automated tests:** Unit tests cover essential functions of the package
   and a reasonable range of inputs and conditions. All tests pass on the local machine.
- [x] **Packaging guidelines**: The package conforms to the rOpenSci packaging guidelines

As far as I can tell, the package mostly conforms to the ROpenSci packaging guidelines.

Here are a few random thoughts from my run-through of the package and read-through of the code:

The files main exported functions — those prefixed with `pmc_` and `separate_` are functional and well-documented, and they parsed XML files I threw at them as I would have expected, given the documentation.

The functions `repeat_sub()` and `path_string()` shouldn't be exported, as they are unrelated to the user-facing functionality and serve internal functions.

The package probably doesn't need the function `pmc_xml()`, which is an alias to `europepmc::epmc_ftxt()`, especially since the README file mention the latter function and not the former.

The main functions are covered by tests.

Code is pretty understandable at a readthrough. There are a few areas where one or two comments might help to make clear the purpose of a block, or a loop, where variable names etc. aren't as understandable at a glance.

There's one "TO DO" comment in `collapse_rows.R` (line 43) — not sure if this is something which is still pending or not.

This is pretty nitpicky now, but, while the code style is consistent throughout, there are one or two areas where spacing before and after assignment operators is a little wonky, or there was less spacing around parentheses and brackets in if/else statements — but again, this is *super* nitpicky and somewhat a matter of personal taste. 😅

Though there are tests to cover most functions, `goodpractice` still flags some lines as not being covered. I don't have that much familiarity with coverage metrics — perhaps there are some cases or outcomes which aren't being tested? I'm not sure.

#### Final approval (post-review)

- [ ] **The author has responded to my review and made changes to my satisfaction. I recommend approving this package.**

Estimated hours spent reviewing: Roughly 4 hours, including setup, reviewing guidelines, and reviewing the package.

—-

### Review Comments

As far as the specific review criteria go, I think this package is really far along in terms of meeting them. A few edits to the documentation, to add a little more context, and resolving the various other minor issues I mentioned, and I think it's good to go.

I think this package also has some clear points of expansion for a next major version. The parsing functions all work very well — but it might be nice to have functions which, for example, extract the body text of an article into a character vector. This sort of thing would do a lot to make the package useful for a larger range of workflows.

NI know I spent most of my writing time here finding nitpicks and critiquing, so I think I should end on a positive note: Nice work! Working with the PMC OAS can be a hassle at times, and this sort of tool
makes it much more accessible.

**Comment by cstubben** [comment]
@toph-allen Thanks for the great suggestions and careful review.  I'll have time later next week to make these changes.  The introduction definitely needs work too

**Comment by juliasilge** [comment]
Here is my review! Thank you 🙏 for your patience as I took a few extra days to get this finished.

## Package Review

This package exhibits good design principles and has been built well to create analysis-ready datasets. 🎉 I think the work put into the API design of the functions is especially excellent.

- [X] As the reviewer I confirm that there are no [conflicts of interest](#coi) for me to review this work (If you are unsure whether you are in conflict, please speak to your editor _before_ starting your review).

#### Documentation

The package includes all the following forms of documentation:

- [ ] **A statement of need** clearly stating problems the software is designed to solve and its target audience in README
- [X] **Installation instructions:** for the development version of package and any non-standard dependencies in README
- [X] **Vignette(s)** demonstrating major functionality that runs successfully locally
- [X] **Function Documentation:** for all exported functions in R help
- [X] **Examples** for all exported functions in R Help that run successfully locally
- [ ] **Community guidelines** including contribution guidelines in the README or CONTRIBUTING, and DESCRIPTION with `URL`, `BugReports` and `Maintainer` (which may be autogenerated via `Authors@R`).

I am an outsider to PubMed and what the target audience for this field of research is, and I think a little bit more context in the README and vignettes would be helpful here. When I visited the GitHub page for [europepmc](https://github.com/ropensci/europepmc), I got a bit more of an understanding of what the problems being solved with this software are.

You might consider switching to [remotes](https://github.com/r-lib/remotes) from devtools for the installation instructions, as remotes is a smaller, more lightweight package without compiled code or dependencies.

For the README and vignettes, I recommend checking out the [tidyverse style guide for pipes](https://style.tidyverse.org/pipes.html) and aligning the way piping is used and formatted with that, or at least so that it is all consistent within this package (pipes go to a new line at the very least, etc). Having a package that explicitly embraces tidy data principles like this one is even better for users when it can align the rest of an ecosystem, in terms of how code is used.

This package is missing a description of contribution guidelines and a code of conduct. This does not need to be long or extensive. You can check [a way that I have written that here](https://github.com/juliasilge/tidytext#community-guidelines). 👥

#### Functionality

- [X] **Installation:** Installation succeeds as documented.
- [X] **Functionality:** Any functional claims of the software been confirmed.
- [X] **Performance:** Any performance claims of the software been confirmed.
- [X] **Automated tests:** Unit tests cover essential functions of the package
   and a reasonable range of inputs and conditions. All tests pass on the local machine.
- [ ] **Packaging guidelines**: The package conforms to the rOpenSci packaging guidelines


Are the functions `repeat_sub()` and `path_string()` meant to be exported functions for the user? It seems like probably not? I would either show how these are meant to be used in a vignette, or else don't export them. Here is [what a bunch of documented but not exported functions](https://github.com/ropensci/qualtRics/blob/master/R/utils.R) look like in a package I maintain, although I forgot to add `#' @noRd` there.

One of the XML files used in a vignette is not small, and causes R CMD check to now give a NOTE:

>❯ checking installed package size ... NOTE
>    installed size is  8.3Mb
>    sub-directories of 1Mb or more:
>      extdata   8.0Mb
>
>0 errors ✔ | 0 warnings ✔ | 1 note ✖

It's PMC6358576_PMC6358715.xml that is the big one, at 7.7Mb. Is it possible to use something smaller? Unfortunately you'll need to get it dramatically smaller to get under 1Mb. 😬 It's not generally good practice to have such large data for vignettes, although I do understand about how useful it can be to use real world data to demonstrate with. I would suggest rethinking this in some way because having this NOTE only for a vignette doesn't seem like a battle worth fighting to get this package on CRAN. I think you should make a smaller dataset to use in vignettes but another options that exists is [precompiling a vignette](https://community.rstudio.com/t/precompiling-vignette-with-devtools/1583/5).

Have you tried out the [styler](https://github.com/r-lib/styler) package to clean up and harmonize the code style of this package? It's pretty painless and can improve the overall readability.

In `DESCRIPTION`, your title should be "Title Case" and also should probably spell out what PMC stands for. Your description should **not** start out with "This package..." Your author field [should look like this](https://github.com/ropensci/qualtRics/blob/master/DESCRIPTION), using the `Authors@R` format.

#### Final approval (post-review)

- [ ] **The author has responded to my review and made changes to my satisfaction. I recommend approving this package.**

Estimated hours spent reviewing: 2.5

---

### Review Comments

It seems like from the title of the package that you are wanting to embrace some tidyverse style and principles. You already have stringr imported in this package, so you might consider using stringr functions like `str_detect()` in place of `grep()` in vignettes, etc. These functions tends to be more readable, more sensible order of arguments, etc.

Also, you may want to consider avoiding loops 🚫 for iteration and instead use purrr, if a goal is to have this package connect to the rest of the suite of tidy tools. For example, in the pmcftp vignette, replace the loop with:

```
tibble(a1) %>%
  mutate(a2 = lead(a1) - 1) %>%
  filter(!is.na(a2)) %>%
  mutate(doc = map2(a1, a2, ~ paste(pmc[.x:.y], 
                                    collapse = "\n")),
         docxml = map(doc, read_xml),
         m1 = map(docxml, pmc_metadata),
         txt1 = map(docxml, pmc_text)) %>%
  transmute(PMCID = map_chr(m1, "PMCID"),
            txt1) %>%
  unnest()
```
  
This code replaces both the chunk called `loop` and the one called `combine` (I have assigned it to anything, but it gives you the idea).

This package is shaping up nicely! ✨ I am really happy to see the excellent work put into this package that will make these datasets so much more easily available to folks who want to access and analyze the content in these texts.

**Comment by noamross** [comment]
Thank you for your reviews, @toph-allen and @juliasilge! @cstubben, please let us know when and how you have addressed the reviewer's comments and if you have any questions.

**Comment by cstubben** [comment]
@juliasilge Thanks for the review and all the helpful links.  I will likely make updates this weekend and use a precompiled vignette to load all 10K articles instead of the 7.7Mb file with 100 articles.

**Comment by cstubben** [comment]
Sorry for the delays, making plans to work on weekends rarely works out.   I tried to address all the suggestions from both reviews and updated the README introduction to fit the statement of need, added community guidelines, updated code using `styler`,  fixed the exports, tidyverse pipe style, PMC abbreviations, title case and the other issues above except two -  I left devtools in the README (never used or installed `remotes`, but maybe I should) and still need to precompile the second FTP vignette and remove the 7.7 Mb file next week.

**Comment by noamross** [comment]
Thank you for the update, @cstubben. @juliasilge and @toph-allen, please let us know if the changes address your comments.

@cstubben I do note that `devtools::install_github()` just calls `remotes::install_github()`, so you have used and installed it. 😉

**Comment by juliasilge** [comment]
This is looking great! I especially think the updated README is improved a lot. 🎉 

From my perspective, things that still definitely need to happen are:

- In the `DESCRIPTION` file, the `Description` field cannot start with "This package..."
- The vignette needs to be updated so it does not use data too large for this package to go to CRAN.

I also still recommend using **remotes** instead of **devtools** for installation instructions, because of how heavy devtools is.

**Comment by cstubben** [comment]
Thanks! I fixed the Description and added **remotes** to the README.   I already reduced the large XML file from the FTP site to 1/10th the previous size, so files in inst/extdata are less than 1 MB combined.

**Comment by juliasilge** [comment]
So @cstubben do you no longer intend to precompile a vignette?

**Comment by cstubben** [comment]
No.  I mainly wanted to show how to find the start of the article nodes and read the combined XML files on the FTP site.  I might use a pre-compiled vignette later to read all 2.5 million articles and look at a few use cases like mentions of "statistically significant" across different journals, but will likely need some collaborators 😄

**Comment by juliasilge** [comment]
From my perspective, this package is ready to be approved! I'm glad to see this sample of the life science published literature so much more accessible. 🙌

**Comment by noamross** [comment]
Thanks @juliasilge. A quick reminder to @toph-allen, who I am aware just returned from a couple of weeks vacation 😉, to take a look at the changes.

**Comment by toph-allen** [comment]
Apologies for my delay. I’m still alive! I’m going to finish going through the updates tomorrow first thing and will post my full response then.

**Comment by toph-allen** [comment]
Sorry it took me a while to finish reviewing the updates.

Looking through the changes, I agree that the package is ready to go! The updated README does a much better job of introducing the topics of the packages, as well as clearly documenting all the package's core functionality. The improvements to the code, both in terms of formatting consistency and minor implementation changes, are also good.

I don't have any additional comments at this time. Nice work! 👏

**Comment by noamross** [comment]
Thank you @toph-allen and @juliasilge!  @cstubben, one tiny thing I found in my final checks: It appears that you have the **europepmc** in `Imports` but it is only used in the vignette, so it should be in `Suggests:`.   With that little fix, we are ready to approve!

To-dos:
- [ ] Transfer the repo to rOpenSci's "ropensci" GitHub organization under "Settings" in your repo.  I have invited you to a team that should allow you to do so.  You'll be made admin once you do.
- [ ] Update any links in badges for CI and coverage to point to the ropensci URL. We no longer transfer Appveyor projects to ropensci Appveyor account so after transfer of your repo to rOpenSci's "ropensci" GitHub organization the badge should be `[![AppVeyor Build Status](https://ci.appveyor.com/api/projects/status/github/ropensci/pkgname?branch=master&svg=true)](https://ci.appveyor.com/project/individualaccount/pkgname)`.
- [ ] Update your `codemeta.json`

Should you want to acknowledge your reviewers in your package DESCRIPTION, you can do so by making them `"rev"`-type contributors in the `Authors@R` field (with their consent).  More info on this [here](https://ropensci.github.io/dev_guide/building.html#authorship).

Our developer guide has a [section](https://ropensci.github.io/dev_guide/collaboration.html) for ongoing maintenance after acceptance. Welcome aboard!

**Comment by cstubben** [comment]
Thanks for all your help!  I fixed the Imports, transferred the repo and updated all GitHub links from cstubben to ropensci.

**Comment by noamross** [comment]
Excellent, all set! Look out for your package website at docs.ropensci.org/tidypmc.  It should be live in a day or so.
